src/main/java/org/embulk/input/zendesk/ZendeskInputPlugin.java

```diff
@@ -0,0 +1,512 @@
+
+        @Min(1)
+        @Max(30)
+        @Config("retry_limit")
+        @ConfigDefault("5")
+        int getRetryLimit();
+
+        @Min(1)
+        @Max(3600)
+        @Config("retry_initial_wait_sec")
+        @ConfigDefault("4")
+        int getRetryInitialWaitSec();
+
+        @Min(30)
+        @Max(3600)
+        @Config("max_retry_wait_sec")
+        @ConfigDefault("60")
+        int getMaxRetryWaitSec();
+
+        @Config("incremental")
+        @ConfigDefault("true")
+        boolean getIncremental();
+
+        @Config("includes")
+        @ConfigDefault("[]")
+        List<String> getIncludes();
+
+        @Config("dedup")
+        @ConfigDefault("true")
+        boolean getDedup();
+
+        @Config("app_marketplace_integration_name")
+        @ConfigDefault("null")
+        Optional<String> getAppMarketPlaceIntegrationName();
+
+        @Config("app_marketplace_org_id")
+        @ConfigDefault("null")
+        Optional<String> getAppMarketPlaceOrgId();
+
+        @Config("app_marketplace_app_id")
+        @ConfigDefault("null")
+        Optional<String> getAppMarketPlaceAppId();
+
+        @Config("columns")
+        SchemaConfig getColumns();
+    }
+
+    private static final Logger logger = Exec.getLogger(ZendeskInputPlugin.class);
+
+    private ZendeskSupportAPIService zendeskSupportAPIService;
+
+    @Override
+    public ConfigDiff transaction(final ConfigSource config, final Control control)
+    {
+        final PluginTask task = config.loadConfig(PluginTask.class);
+        final Schema schema = task.getColumns().toSchema();
+        int taskCount = 1;
+        if (!ZendeskUtils.isSupportIncremental(task.getTarget())) {
+            final JsonNode result = getZendeskSupportAPIService(task).getData("", 0, false, 0);
+            if (result.get(ZendeskConstants.Field.COUNT) != null
+                    && result.get(ZendeskConstants.Field.COUNT).isInt()) {
+                taskCount = ZendeskUtils.numberToSplitWithHintingInTask(result.get(ZendeskConstants.Field.COUNT).asInt());
+            }
+        }
+        return resume(task.dump(), schema, taskCount, control);
+    }
+
+    @Override
+    public ConfigDiff resume(final TaskSource taskSource, final Schema schema, final int taskCount, final Control control)
+    {
+        final PluginTask task = taskSource.loadTask(PluginTask.class);
+        final List<TaskReport> taskReports = control.run(taskSource, schema, taskCount);
+
+        return this.buildConfigDiff(task, taskReports);
+    }
+
+    @Override
+    public void cleanup(final TaskSource taskSource, final Schema schema, final int taskCount, final List<TaskReport> successTaskReports)
+    {
+    }
+
+    @Override
+    public TaskReport run(final TaskSource taskSource, final Schema schema, final int taskIndex, final PageOutput output)
+    {
+        final PluginTask task = taskSource.loadTask(PluginTask.class);
+        ZendeskValidatorUtils.validateInputTask(task, getZendeskSupportAPIService(task));
```

Comment:
Is there reason why `validateInputTask` not in `transaction` instead of? Because `run` will be run in different threads which is not as convenience as main thread in `transaction`